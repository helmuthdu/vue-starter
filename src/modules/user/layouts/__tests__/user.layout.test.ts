import { mount, VueWrapper } from '@vue/test-utils';
import { createMemoryHistory, createRouter } from 'vue-router';
import UserLayout from '../user.layout.vue';

const router = createRouter({ history: createMemoryHistory(), routes: [] });

describe('User/Layout', () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    wrapper = mount(UserLayout, {
      global: {
        plugins: [router],
        stubs: ['router-view'],
      },
    });
  });

  it('should renders with props', () => {
    expect(wrapper.exists()).toBe(true);
  });
});
