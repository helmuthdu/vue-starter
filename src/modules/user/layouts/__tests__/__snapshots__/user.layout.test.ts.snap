// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User/Layout should renders with props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": null,
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <router-view />
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.11",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
  },
  "rootVM": Object {},
  "wrapperElement": <router-view />,
}
`;
